### PR TITLE
Bugfix and clean up for utils.fifo

### PR DESCRIPTION
**Pull request by mjpieters**
Check-list
----------

 * [X] There is a linked issue discussing the motivations for this feature or bugfix
 * [X] Unit tests have been added where possible
 * [ ] ~~I've added / updated documentation for any user-facing features.~~
 * [X] Performance seems to be comparable or better than current `master`

What this change is adding / fixing
-----------------------------------

I split this up into distinct commits to make reviewing easier:

- Add more testing around QuodLibetUnixRemote and FIFO
- Fix FIFO._process error handling
- Clean up FIFO open/process handing
- Modernize exception handling
- Final clean-up of remaining issues

This addresses #3745, and should leave the code in an easier-to-maintain
state.

### CODE REVIEW

File: tests/test_remote.py [hunk 1]
```diff
@@ -41,3 +46,106 @@ def test_response(self):
...
+            yield
+
+    def _send_message_remote_proc(self, msg, callback):
+        """Execute QuodLibetUnixRemote.send_message() in a child process"""
+        # Using a child process is the only way to execute send_message, as it
+        # can only be run on the main thread, and blocks waiting for the
+        # response. It can't be run in a separate thread, because it uses
+        # signals to handle timeouts and we can't run the GLib mainloop in a
+        # thread as Gtk has already been initialised on the Python main thread.
+        # Attempting to use the loop in a thread leads to segfaults.
```
Comment (mjpieters) [inline tests/test_remote.py:67]: I tried the thread approach first, but that segfaults in `gdk_event_prepare()` on macos. That's not that surprising given the explicit warnings against this in the [GDK threading documentation](https://developer.gnome.org/gdk3/stable/gdk3-Threads.html):

> GTK+, however, is not thread safe. You should only use GTK+ and GDK from the thread `gtk_init()` and `gtk_main()` were called on. This is usually referred to as the “main thread”.

Before the test is even run, `quodlibet._init()` has already called `Gtk.init_args()`.

**Comment by mjpieters** [comment]
Weird, I had the tests succeed in a Docker container, locally. I'll investigate why the fedora and ubuntu tests failed on GH.

**Comment by mjpieters** [comment]
Source of the test failure located (no virtualenv meant the main quodlibet package was not being found in the child process).

**Comment by declension** [comment]
Thanks!

**Pull request merged**